src/PhpUnit/AssertsTrait.php

```diff
@@ -206,8 +214,18 @@ public function assertRequestQueryMatch(
             // @codeCoverageIgnoreEnd
         }
 
-        $constraint = new RequestQueryConstraint($schemaManager->getRequestQueryParameters($template, $httpMethod));
+        $constraint = new RequestQueryConstraint($schemaManager->getRequestQueryParameters($template, $httpMethod), $this->getValidator());
 
         Assert::assertThat($query, $constraint, $message);
     }
+
+    /**
+     * Returns a new Validator instance.
+     *
+     * @return Validator
+     */
+    protected function getValidator()
+    {
+        return new Validator(Constraint::CHECK_MODE_NORMAL, null, $this->constraintFactory);
```

Comment:
This should be `return new Validator();`, custom behaviors are done by "overriding" this method.